src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Extensions/AssemblyExtensions.cs

```diff
@@ -13,12 +13,20 @@ public static class AssemblyExtensions
         /// Loads the <see cref="Assembly"/>'s <see cref="Type"/>s that can be loaded ignoring others.
         /// </summary>
         /// <param name="assembly"><see cref="Assembly"/> instance.</param>
+        /// <param name="includeReferenced">Set to true to load <see cref="Type"/>s from referenced assemblies too</param>
         /// <returns>Returns the list of <see cref="Type"/>s that can be loaded.</returns>
-        public static Type[] GetLoadableTypes(this Assembly assembly)
+        public static Type[] GetLoadableTypes(this Assembly assembly, bool includeReferenced = false)
```

Comment:
I made this method to include all referencing assemblies and ran a sample app to apply the logic for the in-proc worker scenario. Then, it started complaining.

I haven't started testing only for the out-of-worker scenario yet. Are you able to run both in-proc and out-of-proc worker scenarios on your end?

Reply:
I remember my first edit was to make the method return all referenced types as well by default and saw the complaints, that's why I introduced it as an optional parameter and changed the invocation in only one place; the place where it looks for the attribute for the out-of-process worker. There should be a similar call from the in-process worker where you may change the call as well without impact. 
I didn't try to debug or see why it was complaining in the other places it was called as I didn't have enough time and thought you might know as soon as you saw it. I can have another look when I get some time

Reply:
Hi @justinyoo , I had a deeper look and it turns out that the change causes errors for 2 reasons:
1) The ```VisitorCollection.CreateInstance()``` method looks for types that end with "Visitor". If we load referenced assemblies this causes problems as other assemblies may have types that end in Visitor and are not compliant with the logic (e.g. they don't have parameterless constructors or any public constructors at all etc. Such is the case for a NewtonsoftJson package that is included in the referenced assemblies)
2) Loading referenced assemblies causes problems with the ```.SingleOrDefault(...)``` calls that expect for example one implementation of ```IOpenApiConfigurationOptions```. By loading the referenced assemblies it loads the potential client function's implementation along with the referenced ```DefaultOpenApiConfigurationOptions``` and ```OpenApiSettings``` of your project.

To resolve (1) I changed the code in ```VisitorCollection.CreateInstance()``` to look for IVisitor interface implementations instead of classes ending in "Visitor", which should be ok, as the linq query has in its select statement ```(IVisitor)Activator.CreateInstance...``` so if it's not ```IVisitor``` there would be a problem anyway.

To resolve (2) I just excluded all types that belong in a namespace that starts with ```Microsoft.Azure.WebJobs.Extensions.OpenApi```. That excludes all of your project so it works fine.

Now the ```GetLoadableTypes(...)``` can always include referenced assemblies and it seems to be working for me.